app/uk/gov/hmrc/breathingspaceifproxy/controller/service/RequestValidation.scala

```diff
@@ -44,7 +49,7 @@ trait RequestValidation {
       validateContentType(request),
       validateCorrelationId(headers),
       validateRequestType(headers, endpointId),
-      validateStaffPid(headers)
+      validateStaffPid(headers, endpointId)
```

Comment:
Not too keen on this approach. It is tying the low level details with the higher level details when it isn't required as we do not need to check staff pid or request type in the new api endpoint.

What I would do is check the endpoint in this method and only compose the methods required for that endpoint. 
```
def validateHeadersForNPS(endpointId: EndpointId, upstreamConnector: UpstreamConnector)(
    implicit request: Request[_]
  ): Validation[RequestId] = {
    endpointId match {
      case BS_Memorandum_GET => validateMemorandumHeaders(endpointId, upstreamConnector)
      case _                 => validateAllHeaders(endpointId, upstreamConnector)
    }
  }

  private def validateAllHeaders(endpointId: EndpointId, upstreamConnector: UpstreamConnector)(
    implicit request: Request[_]
  ): Validation[RequestId] = {
    val headers = request.headers
    (
      validateContentType(request),
      validateCorrelationId(headers),
      validateRequestType(headers, endpointId),
      validateStaffPid(headers, endpointId)
    ).mapN((_, correlationId, attended, staffPid) => (correlationId, attended, staffPid))
      .andThen(validateStaffPidForRequestType(endpointId, upstreamConnector))
  }
  
  private def validateMemorandumHeaders(endpointId: EndpointId, upstreamConnector: UpstreamConnector)(
    implicit request: Request[_]
  ): Validation[RequestId] = {
    val headers = request.headers
    (
      validateContentType(request),
      validateCorrelationId(headers)
    ).mapN((_, correlationId) => (correlationId, DA2_PTA, unattendedStaffPid))
      .andThen(validateStaffPidForRequestType(endpointId, upstreamConnector))
  }
```

This is still not ideal as the `RequestId` requires us to specify an attended status when it is not needed and a staffpid when it is not needed, there is also an upstream, connector provided which is awful design but I will raise a ticket to tackle these issues. if you make this change you won't need to make changes to the other methods which will make testing simpler and the logic easier to digest imo